audit.py

```diff
@@ -715,5 +724,239 @@ def members(
+        with self.tag("td"):
+            self.text(rule.get("security_severity_level") or "")
+        with self.tag("td"):
+            self.text(rule.get("severity") or "")
+        self.doc.stag("td")
+        with self.tag("td"):
+            self.text(most_recent.get("ref") or "")
+        with self.tag("td"):
+            self.text(most_recent.get("state") or "")
+
+
+def convert_alert_instance(instance):
+    converted = {
+        "ref": instance.ref,
+        "state": instance.state,
+        "commit_sha": instance.commit_sha,
+        "location": {
+            "path": instance.location.path,
+            "start_line": instance.location.start_line,
+            "start_column": instance.location.start_column,
+            "end_line": instance.location.start_line,
+            "end_column": instance.location.start_column,
+        },
+        "analysis_key": instance.analysis_key,
+        "message": instance.message.get('text')
+    }
+    return converted
+
+
+JSON_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S"
+
+
+def convert_alert(alert, verbose):
+    converted = {
+        "number": alert.number,
+        "created_at": alert.created_at.strftime(JSON_DATETIME_FORMAT) if alert.created_at else None,
+        "dismissed_at": alert.dismissed_at.strftime(JSON_DATETIME_FORMAT) if alert.dismissed_at else None,
+        "tool": {
+            "name": alert.tool.name,
+            "version": alert.tool.version,
+        },
+        "rule": {
+            "name": alert.rule.name,
+            "description": alert.rule.description,
+            "security_severity_level": alert.rule.security_severity_level,
+            "severity": alert.rule.severity,
+        },
+        "most_recent_instance": convert_alert_instance(alert.most_recent_instance),
+    }
+    if verbose:
+        converted["instances"] = [convert_alert_instance(instance) for instance in alert.get_instances()]
+    return converted
+
+
+def empty_alert():
+    converted = {
+        "number": "",
+        "created_at": None,
+        "dismissed_at": None,
+        "tool": {},
+        "rule": {},
+        "most_recent_instance": {},
+        "instances": []
+    }
+    return converted
+
+
+if using_github_package:
+    def get_codescanning_alerts_for_repo(repo, verbose):
+        return [
+            convert_alert(alert, verbose)
+            for alert in repo.get_codescan_alerts()
+        ]
+else:
+    requester = Requester(
+        login_or_token=config["github.com"]["token"],
+        password=None,
+        jwt=None,
+        base_url="https://api.github.com",
+        timeout=15,
+        user_agent="PyGithub/Python",
+        per_page=30,
+        verify=True,
+        retry=None,
+        pool_size=None,
+    )
+
+    def get_codescanning_alerts_for_repo(repo, verbose):
+        return [
+            convert_alert(alert, verbose)
+            for alert in PaginatedList(CodeScanAlert, requester, f"{repo.url}/code-scanning/alerts", {})
+        ]
```

Comment:
See comment above at the import statement. I would expect the CodeScanAlert invocation to be the same whether the GitHub package is used or the local implementation?

Reply:
Added comment to clarify